src/transformers/models/mixtral/modeling_mixtral.py

```diff
@@ -837,9 +837,21 @@ def __init__(self, config):
 
         self.experts = nn.ModuleList([MixtralBlockSparseTop2MLP(config) for _ in range(self.num_experts)])
 
+        # Jitter parameters
+        self.jitter_noise = config.router_jitter_noise
+
+    def jitter(self, x):
+        if self.training and self.jitter_noise > 0:
+            low = 1.0 - self.jitter_noise
+            high = 1.0 + self.jitter_noise
+            noise = torch.rand(x.size(), dtype=x.dtype, device=x.device)
+            return x * (low + noise * (high - low))
+        return x
```

Comment:
no need for a new function, let's do something like https://github.com/huggingface/transformers/blob/b345f4e7f7f852f7def0ebe473d21419ae235fdc/src/transformers/models/switch_transformers/modeling_switch_transformers.py#L163